vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -77,7 +79,23 @@ func NewVolumeDriver(port int, useMockEsx bool, mountDir string, driverName stri
 		}
 	}
 
-	d.refCounts.Init(d, mountDir, driverName)
+	refCountStatus := d.refCounts.Init(d, mountDir, driverName)
+	// If refcounting wasn't successfull, schedule one again
+	if refCountStatus == false {
+		timer := time.NewTimer(refCountDelay)
+		log.Infof("Refcounting failed. Scheduling again after %s seconds", refCountDelay)
+		go func() {
+			<-timer.C
+			refCountStatus = d.refCounts.Init(d, mountDir, driverName)
+			if refCountStatus {
+				log.Infof("Refcounting reattempt succeded")
+			} else {
+				log.Infof("Refcounting reattempt failed")
```

Comment:
This should be a warning log. Preferably have an error status returned vs. a boolean so the user knows exactly why the refcount recovery failed.

There's no recovery if this attempt fails and volumes will be left attached to the VM till the VM is powered off and the other fix comes into play to remove volumes still attached to the VM. Which can be a long time. Do we want to do the recovery a fixed number of times before giving up. Like, docker may have a problem and may not be up for some time and the retry will be useless in that time period. A few fixed retries may be good.